src/calendar/model/CalendarModel.ts

```diff
@@ -242,6 +244,10 @@ export class CalendarModel {
 	private async getCalendarDataForUpdate(fileId: IdTuple): Promise<ParsedCalendarData | null> {
 		try {
 			const file = await this.entityClient.load(FileTypeRef, fileId)
+			if (file._ownerEncSessionKey == null) {
+				// owner enc session key not updated yet - see NoOwnerEncSessionKeyForCalendarEventError's comment
+				return Promise.reject(new NoOwnerEncSessionKeyForCalendarEventError("no owner enc session key found on the calendar data's file"))
```

Comment:
should this just throw the error? the interaction with the try block and the catch below is unclear

Reply:
It could. The IDE gives a pretty unsightly warning if you do this though, which is why it was changed to a Promise.reject.

Reply:
fair enough, I tend to just ignore that warning. maybe we should disable it completely because it can't recognize `catch` clauses that are meant to catch other error types and rethrow everything else.
in the future, `// noinspection ExceptionCaughtLocallyJS` is an option for this as well.